src/main/java/seedu/duke/model/item/Link.java

```diff
@@ -2,7 +2,13 @@
 
 package seedu.duke.model.item;
 
+import seedu.duke.DukeException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Link extends Item {
+    public static final Pattern MODULE_CODE_PATTERN = Pattern.compile("(^[A-Z]{2,3}[\\d]{4}[A-Z]?$)");
```

Comment:
Consider using the static constant from `Module` class instead of rewriting the code here, so that this code doesn't need to be duplicated and can be changed from 1 single place.

Reply:
Thanks for the suggestion! I've refactored it.